internal/handlers/common/filter.go

```diff
@@ -333,6 +336,41 @@ func filterOperator(doc *types.Document, operator string, filterValue any) (bool
 	}
 }
 
+// filterExprOperator uses $expr aggregation operator to process the document
+// and returns boolean indicating filter match.
+//
+// If the result of processing $expr is null, zero value or false, it returns
+// false indicating filter was not matched. For other values, it returns true.
```

Comment:
Ok, then don't understand why after having `v, err := op.Process(doc)` we can't just return `v` as is but need to do some extra type switch :) Let's maybe take a look after the daily sync, as I really want to understand the difference between `process` and `filter`.

Reply:
It's a strange behaviour for sure. I found no documentation of it why it behaves such way. My guess is mongoDB is mapping each value to `truthy` or `falsy` values. `null`, `zero` and `false` appears to be `falsy` but why does that do that :sweat_smile: 

With `task run-proxy` and `task mongosh` let's see that it does.
The `truthy` values like `1`, `true` find something and `falsy` values like `null` `0` and `false` doesn't find.
```
test> db.Nulls.find({$expr: 1})
[ { _id: 'null', v: null } ]
test> db.Nulls.find({$expr: true})
[ { _id: 'null', v: null } ]
test> db.Nulls.find({$expr: null})

test> db.Nulls.find({$expr: 0})

test> db.Nulls.find({$expr: false})

```

Now using expression instead of constant on different collection, it finds value that is not `falsy`.
```
test> db.Unsets.find()                
[ { _id: 'unset' } ]
test> db.Unsets.find({$expr: "$v"})
                                                   
test> db.Nulls.find()         
[ { _id: 'null', v: null } ]     
test> db.Nulls.find({$expr: "$v"}) 
                                                   
test> db.Int32s.find()        
[                                     
  { _id: 'int32-1', v: 4080 },        
  { _id: 'int32-2', v: 1048560 },
  { _id: 'int32-3', v: 268435440 },
  { _id: 'int32', v: 42 },
  { _id: 'int32-zero', v: 0 },
  { _id: 'int32-max', v: 2147483647 },
  { _id: 'int32-min', v: -2147483648 }
]
test> db.Int32s.find({$expr: "$v"})
[
  { _id: 'int32-1', v: 4080 },
  { _id: 'int32-2', v: 1048560 },
  { _id: 'int32-3', v: 268435440 },
  { _id: 'int32', v: 42 },
  { _id: 'int32-max', v: 2147483647 },
  { _id: 'int32-min', v: -2147483648 }
]
```